packages/vite/src/node/shortcuts.ts

```diff
@@ -119,3 +121,83 @@ const BASE_SHORTCUTS: CLIShortcut[] = [
     },
   },
 ]
+
+type PreviewShortcuts = {
+  key: string
+  description: string
+  action: (server: PreviewServer) => void
+}
+
+export function bindPreviewShortcuts(server: PreviewServer): void {
```

Comment:
Let's factorise this code a bit with something like that:

```ts
type CLIShortcut<Server> = {
  key: string
  description: string
  action: (server: Server) => void
}

function bindShortcuts<Server>(server: Server, shortcuts: CLIShortcut<Server>[]): void {
  // almost exactly this code below
}

export function bindPreviewShortcuts(server: PreviewServer): void {
  bindShortcuts(server, previewShortcuts)
}

type PreviewShortcut = CLIShortcut<PreviewServer>
```

And then rename the other usage in this file to add a dev prefix